python/ccxt/base/exchange.py

```diff
@@ -361,10 +348,20 @@ def define_rest_api(cls, api, method_name, options={}):
                         if 'underscore' in options['suffixes']:
                             underscore += options['suffixes']['underscore']
 
-                    args = [url, api_type, uppercase_method]
-                    partial_descriptor = PartialDescriptor(entry, args, {})
-                    setattr(cls, camelcase, partial_descriptor)
-                    setattr(cls, underscore, partial_descriptor)
+                    def partialer():
+                        outer_kwargs = {'path': url, 'api': api_type, 'method': uppercase_method}
+
+                        @functools.wraps(entry)
+                        def inner(*args):
+                            inner_kwargs = dict(outer_kwargs)  # avoid mutation
+                            if len(args) > 1:
+                                inner_kwargs['params'] = args[1]
+                                args = [args[0]]  # a reference to self without circular dependencies
+                            return entry(*args, **inner_kwargs)
+                        return inner
+                    to_bind = partialer()
+                    setattr(cls, camelcase, to_bind)
+                    setattr(cls, underscore, to_bind)
         cls.rest_api_defined = True
         # ^ this code only runs once per class
```

Comment:
Will lose the capability to add rest api at runtime?

Reply:
You can still modify the rest api at runtime, but only by decorating (wrapping with another function) `.describe` before initiating any instances. 

```
exchange_class = ccxt.bittrex()
exchange_class.describe = decorator(exchange_class.describe)
exchange_class()  # no more rest api's can be defined after this
```

but I think you'd be better off using the descriptors or binding your own custom rest api functions than trying to do it like this